app/models/user.rb

```diff
@@ -45,25 +47,15 @@ def name
     "#{first_name} #{last_name}"
   end
 
-  def approved_accounts
-    Account.
-      joins('LEFT OUTER JOIN `memberships` ON `memberships`.`account_id` = `accounts`.`id`').
-      where(['memberships.user_id = ? and memberships.approved is true', id])
-  end
-
   def approved_active_accounts
-    approved_accounts.active
+    approved_accounts.try(:active)
   end
 
   def approved_account_stories
-    Story.
-      joins('LEFT OUTER JOIN `memberships` ON `memberships`.`account_id` = `pieces`.`account_id`').
-      where(['memberships.user_id = ? and memberships.approved is true', id])
+    !approved_accounts.nil? ? Story.where(account_id: approved_accounts.try(:ids)) : []
```

Comment:
could `Story.where(account_id: approved_accounts.try(:ids))`  be in a controller method instead of in the user model? doesn't seem to rely be using anything in the user model, except the convenience that it is the `resource` loaded by authorized controllers.